scripts.py

```diff
@@ -292,12 +302,59 @@ def __init__(self, provider_class, _db=None, providers=None, **kwargs):
         providers = providers or list()
         if provider_class:
             providers += self.get_providers(_db, provider_class, **kwargs)
-        super(RunCollectionCoverageProviderScript, self).__init__(providers)
+        super(RunCollectionCoverageProviderScript, self).__init__(providers, _db=_db)
 
     def get_providers(self, _db, provider_class, **kwargs):
         return list(provider_class.all(_db, **kwargs))
 
 
+class RunThreadedCollectionCoverageProviderScript(Script):
+    """Run coverage providers in multiple threads."""
+
+    DEFAULT_WORKER_SIZE = 5
+
+    def __init__(self, provider_class, worker_size=None, _db=None, **provider_kwargs):
+        super(RunThreadedCollectionCoverageProviderScript, self).__init__(_db)
+
+        self.session_factory = SessionManager.sessionmaker(session=self._db)
+        self.worker_size = worker_size or self.DEFAULT_WORKER_SIZE
+
+        self.provider_class = provider_class
+        self.provider_kwargs = provider_kwargs
+
+    def worker_factory(self, job_queue):
+        """Returns a Worker with a database"""
+        return DatabaseWorker(job_queue, scoped_session(self.session_factory))
+
+    def run(self):
+        _db = scoped_session(self.session_factory)
```

Comment:
I don't understand scoped_session well enough to know if this is a problem, but I'm concerned that you're calling scoped_session on the same factory twice, once to make the DatabaseWorker and once within run(). Will both of these calls give the same scoped session or will there be two different scoped sessions?

Reply:
So, this is ultimately working because I'm creating different a new `scoped_session` for each thread, but yeah, it's not doing what I had hoped it was. I'm going to try to adjust it to use the factory directly, since scoping isn't giving me anything.